File: mmi-java/src/br/ufscar/lince/mmi/MMIEvent.java
package br.ufscar.lince.mmi;

public class MMIEvent extends JWrapper {
	
	protected MMIEvent(long pointer) {
		this.pointer = pointer;
	}
	
	public void finalize() {
		destroy();
	}
	
	public native String getDeviceId();
	
	public native String getEventType();
	
	private native void destroy();
	
	static public MMIEvent CovertEvent(MMIEvent event) throws Exception {
		throw new Exception("You should'nt call this method with an superclass MMIEvent " +
				"instance. Use its subclasses instead");
	}
}

File: src/MMIEvent.cpp
/*
 * MMIEvent.cpp
 *
 *  Created on: Sep 24, 2010
 *      Author: caioviel
 */

#include "../include/MMIEvent.h"

namespace br {
namespace ufscar {
namespace lince {
namespace mmi {

MMIEvent::MMIEvent(std::string deviceId, std::string eventType) {
	this->deviceId = deviceId;
	this->eventType = eventType;
}

MMIEvent::~MMIEvent() {

}

std::string MMIEvent::getDeviceId() {
	return deviceId;
}

std::string MMIEvent::getEventType() {
	return eventType;
}

}
}
}
}
